src/lib.rs

```diff
@@ -16,6 +16,12 @@
 //! suffice, but sometimes an annotation is required, e.g.
 //! `rand::random::<f64>()`.
 //!
+//! Note that these randomly generated values are not necessarily uniform over
+//! *all* possible values of the type, just to a reasonable amount. For example,
+//! `rand::random::<f64>()` will never return `f64::NAN`, and will only return
+//! a value in the range [0, 1). `Option<u8>` will have a 1/2 chance of being,
+//! `None`, not 1/257.
```

Comment:
It is widely accepted/known that floating point random numbers are generated in range 0 to 1. As for `None`, if you ignore the `NonNull` optimisation, it having a chance of ½ is uniform from the perspective of bit pattern, which is (discriminant, value).